crates/gpui/src/platform/blade/blade_renderer.rs

```diff
@@ -428,29 +429,31 @@ impl BladeRenderer {
             path_intermediate_msaa_texture,
             path_intermediate_msaa_texture_view,
             rendering_parameters,
+            skip_draws: false,
         })
     }
 
-    fn wait_for_gpu(&mut self) {
-        if let Some(last_sp) = self.last_sync_point.take()
-            && !self.gpu.wait_for(&last_sp, MAX_FRAME_TIME_MS)
-        {
-            log::error!("GPU hung");
-            #[cfg(target_os = "linux")]
-            if self.gpu.device_information().driver_name == "radv" {
-                log::error!(
-                    "there's a known bug with amdgpu/radv, try setting ZED_PATH_SAMPLE_COUNT=0 as a workaround"
-                );
+    fn wait_for_gpu(&mut self) -> anyhow::Result<()> {
+        if let Some(last_sp) = self.last_sync_point.take() {
+            if !self.gpu.wait_for(&last_sp, MAX_FRAME_TIME_MS) {
+                log::error!("GPU hung");
+                #[cfg(target_os = "linux")]
+                if self.gpu.device_information().driver_name == "radv" {
+                    log::error!(
+                        "there's a known bug with amdgpu/radv, try setting ZED_PATH_SAMPLE_COUNT=0 as a workaround"
+                    );
+                    log::error!(
+                        "if that helps you're running into https://github.com/zed-industries/zed/issues/26143"
+                    );
+                }
                 log::error!(
-                    "if that helps you're running into https://github.com/zed-industries/zed/issues/26143"
+                    "your device information is: {:?}",
+                    self.gpu.device_information()
                 );
+                return Err(anyhow::anyhow!("GPU device hung or lost"));
```

Comment:
Actually, it depends on wait_for implementation, could be just a normal frame timeout. Although, MAX_FRAME_TIME_MS is set to 10 seconds, so it usually coincides. The major architectural issue is that blade doesn't surface error codes to the application layer, so it's impossible to know whether device is really lost. But this is a good start, although it might lead to false positives on some platforms.